src/Features/Core/Portable/RemoveAsyncModifier/AbstractRemoveAsyncModifierCodeFixProvider.cs

```diff
@@ -0,0 +1,236 @@
+            var node = token.GetAncestor(IsAsyncSupportingFunctionSyntax);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+            var methodSymbol = GetMethodSymbol(node, semanticModel, cancellationToken);
+
+            if (ShouldOfferFix(methodSymbol, knownTypes))
+            {
+                context.RegisterCodeFix(
+                    new MyCodeAction(c => FixAsync(document, diagnostic, c)),
+                    context.Diagnostics);
+            }
+        }
+
+        protected sealed override async Task FixAllAsync(
+            Document document, ImmutableArray<Diagnostic> diagnostics,
+            SyntaxEditor editor, CancellationToken cancellationToken)
+        {
+            var generator = editor.Generator;
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var compilation = await document.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+            var knownTypes = new KnownTypes(compilation);
+
+            foreach (var diagnostic in diagnostics)
+            {
+                var token = diagnostic.Location.FindToken(cancellationToken);
+                var node = token.GetAncestor(IsAsyncSupportingFunctionSyntax);
+                var methodSymbol = GetMethodSymbol(node, semanticModel, cancellationToken);
+
+                RemoveAsyncModifier(editor, semanticModel, node, methodSymbol, knownTypes);
+            }
+        }
+
+        private static IMethodSymbol GetMethodSymbol(SyntaxNode node, SemanticModel semanticModel, CancellationToken cancellationToken)
+        {
+            if (semanticModel.GetSymbolInfo(node, cancellationToken).Symbol is IMethodSymbol methodSymbol)
+            {
+                return methodSymbol;
+            }
+
+            return semanticModel.GetDeclaredSymbol(node, cancellationToken) as IMethodSymbol;
+        }
+
+        private void RemoveAsyncModifier(SyntaxEditor editor, SemanticModel semanticModel, SyntaxNode originalNode, IMethodSymbol methodSymbol, KnownTypes knownTypes)
+        {
+            // SyntaxEditor can't change or remove tokens so we have to replace the whole node for the method
+            // which means this edit must come first
+            var replacementNode = RemoveAsyncModifier(methodSymbol, originalNode, knownTypes);
+            editor.ReplaceNode(originalNode, replacementNode);
+
+            // Now that we've replaced the original node we have to work on the replacement for future work
+            if (TryGetExpressionBody(replacementNode, out var expressionBody))
+            {
+                if (methodSymbol.ReturnType == knownTypes._taskType)
+                {
+                    // We need to add a return so we have to convert to a block body
+                    var blockBodiedNode = ConvertToBlockBody(replacementNode, expressionBody, editor);
```

Comment:
nit: try to keep parameters in "scoping order".  i.e. your method had the editor as it's first parameter.  so it should be passed in before any locals you computed in here.  it helps keep a natural ordering across all related membesr.